specifyweb/frontend/js_src/lib/components/Forms/uiFormatters.ts

```diff
@@ -175,6 +175,10 @@ abstract class Field {
   public valueRegexp(): string {
     throw new Error('not implemented');
   }
+
+  public getSize(): number {
```

Comment:
I was thinking of that, but wasn't sure if it is a convention thing with object oriented programming to rely more on methods or not. Will change size to just public

Reply:
in Java, the convention would be to not expose properties directly, but to expose getters/setters

the thinking is that you have more control in that case, and can add extra validation logic, without having to refactor the usage

but, this is one of the things that gives Java it's reputation for verbosity and boilerplate (lol, I read a report claiming that GitHub Copilot is most helpful for Java developers due to how much boiplerpalce there is, with up to 51% of code being written by Copilot)

In JavaScript, there are unfortunately fewer clear standards, but I believe most prefer brevity when it comes to cases like this.
And also, modern refactoring tools in IDEs + TypeScript typechecking makes refactoring code much easier in case you would have to change in the future, reducing the need for premature overcomplication

Reply:
oh, and actually even if you did have to add some custom logic to the getter of this property in the future, you can use native JavaScript getters rather than your custom method:

```js
  private _size: number;
  public get size(): number {
    // some custom logic here
    return this._size;
  }
```

replacing a property with a getter wouldn't require refactoring the usages at all